enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,654 @@
+    // Note that this option only applies to cleartext HTTP connections
+    // and to secure HTTP connections for which the ingress controller
+    // terminates encryption (that is, edge-terminated or reencrypt
+    // connections).  Headers cannot be set(replace)/append for TLS passthrough
+    // connections.
+    //
+    // +kubebuilder:validation:Optional
+    // +optional
+    Set *IngressControllerSetHTTPHeaders `json:"set,omitempty"`
+
+    // delete defines HTTP headers that should be deleted.
+    // If this field is empty, no headers are deleted.
+    //
+    // Note that this option only applies to cleartext HTTP connections
+    // and to secure HTTP connections for which the ingress controller
+    // terminates encryption (that is, edge-terminated or reencrypt
+    // connections).  Headers cannot be set(replace)/append for TLS passthrough
+    // connections.
+    //
+    // +kubebuilder:validation:Optional
+    // +optional
+    Delete *IngressControllerDeleteHTTPHeaders `json:"delete,omitempty"`
+}
+```
+
+The IngressControllerSetHTTPHeaders type has a Response field,
+of type []IngressControllerSetHTTPHeader, for specifying headers to set/append:
+
+```go
+// The IngressControllerSetHTTPHeaders type has a Response field,
+// type []IngressControllerSetHTTPHeader, for specifying headers to set(replace) or append:
+
+type IngressControllerSetHTTPHeaders struct {
+    // response specifies which HTTP response headers to set(replace) or append.
+    //
+    // If this field is empty, no response headers are set(replace) or append.
+    //
+    // +nullable
+    // +optional
+    Response []IngressControllerSetHTTPHeader `json:"response,omitempty"`
+}
+```
+
+The IngressControllerSetHTTPHeader type has fields for specifying a header name, header value and action.
+```go
+type IngressControllerSetHTTPHeader struct{
+    // name specifies a header name.  Its value must be a valid HTTP header 
+    // name as defined in RFC 2616 section 4.2.
+    //	
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=256
+    // +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$" 
+    // +required 
+    Name string `json:"name"`
+
+    // Value specifies a header value.
+    // +kubebuilder:validation:Required 
+    // +required
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=4096
+    Value string `json:"value"`
+
+    // Action specifies a header value.
+    // +kubebuilder:validation:Required
+    // +required
+    // +kubebuilder:validation:Enum=Append;Replace
+    Action IngressControllerHTTPHeaderAction `json:"action"`
+}
+```
+
+This defines Actions to be perfomed on headers when action field is `Set` in the IngressControllerHTTPHeaders. 
+```go
+// IngressControllerHTTPHeaderAction is a policy for setting HTTP headers.
+// +kubebuilder:validation:Enum=Append;Replace
+type IngressControllerHTTPHeaderAction string
```

Comment:
This is not accurate anymore, you use `Set` and `Delete` now.

Reply:
For setting headers  we either Append or Replace(Set). So we need this.

Reply:
Sorry confused it with a similar type which for some reason is not (but should) present in the EP: `IngressControllerHTTPHeaderAction`.

Reply:
api changes are updated